### DISCUSSION TITLE
Is there a way to force height and width of video remote tracks?

### DISCUSSION THREAD

**Discussion by asticode** [pion/webrtc#2206] [General] [answered]
Hi there,

I was wondering whether it was possible to force height and width on video remote tracks? Something similar to `rtcp.ReceiverEstimatedMaximumBitrate` which, if I understand correctly, allows to set the maximum bitrate the remote video track can contribute in.

Cheers

**Comment by Sean-Der** [comment, marked as answer]
@asticode Unfortunately there is nothing you can do as the receiver to force/request a certain resolution. On the client side you can configure it was [Media Stream Constraints](https://developer.mozilla.org/en-US/docs/Web/API/Media_Streams_API/Constraints)

You may not even need `rtcp.ReceiverEstimatedMaximumBitrate`. By default you should get bandwidth probing from the Feedback Reports (`Transport Wide Congestion Control Reports`) and then setting the constraints should get you what you need.

Another thing you should check out is Simulcast! It allows you to upload 3 video streams and you pick the resolution (low, med and high) which might be helpful if you are trying to get a specific resolution.

**Reply by alexpokotilo** [reply]
> You may not even need `rtcp.ReceiverEstimatedMaximumBitrate`. By default you should get bandwidth probing from the Feedback Reports (`Transport Wide Congestion Control Reports`) and then setting the constraints should get you what you need.
> 
@Sean-Der, thanks for your answers. Could you please explain this method or point to any code/example? Thanks in advance.

**Reply by Sean-Der** [reply]
@alexpokotilo Yea! See https://github.com/pion/webrtc/discussions/2206#discussioncomment-2708053, happy to add any more details that could be helpful.

**Reply by alexpokotilo** [reply]
@Sean-Der thanks. I get your point now. jsfiddle example shows how to set exact video dimensions and it is clear. But I though I should do something to process "Feedback Reports (Transport Wide Congestion Control Reports)" explicitly in my code, but looking into save-to-disk I was not able to find anything there. It looks like I get your reply below wrong: 

> By default you should get bandwidth probing from the Feedback Reports (Transport Wide Congestion Control Reports)  and then setting the constraints should get you what you need.

I thought I need to to add some "constrains" into my custom reply to "Feedback Reports". But you actually meant I need to set constraints for getUserMedia and no change required in pion part.
Sorry for misunderstanding

**Comment by asticode** [comment]
First off, thanks for making me discover Github discussions, I didn't even know this was a thing, the tool seems pretty cool 👍

> By default you should get bandwidth probing from the Feedback Reports (Transport Wide Congestion Control Reports) and then setting the constraints should get you what you need.

Is there an example that would show me how to properly set this up?

**Reply by Sean-Der** [reply]
I really like the discussions! I hope that it is a good resource for others to find/use. We will see :)

> Is there an example that would show me how to properly set this up?

Check this out! Run this `jsfiddle` against the `save-to-disk` example. Instead of just getting the best possible resolution it forces 640x480 [here](https://jsfiddle.net/o043k9u6/1/)
